packages/package-json-lint/src/rules/require-no-hard-coded-dependency-versions.js

```diff
@@ -0,0 +1,23 @@
+module.exports = {
+  create: ({ ruleConfig, projectType, report }) => ({
+    dependencies: (dependencies) => {
+      if (projectType === 'module' || projectType === 'devModule') {
+        const messageString = 'no hard coded dependency';
+        const currentProblems = Object.keys(dependencies).map(dependencyName => {
+          const dependencyVersion = dependencies[dependencyName];
+          if (!dependencyVersion.startsWith('^') && !(ruleConfig.severity.allowList && ruleConfig.severity.allowList.includes(dependencyName))) {
```

Comment:
@ryanthemanuel did you have a suggestion for a more robust version check?

Reply:
I have added a regex array to check for compatibility versions here https://github.com/cerner/terra-toolkit/pull/704/commits/b3446c5a790a1172c0c86da818d7d21076835b53.